packages/yum.go

```diff
@@ -185,15 +157,37 @@ func YumUpdates(opts ...YumUpdateOption) ([]PkgInfo, error) {
 			err = nil
 		}
 	}
+
 	// Since we don't get good error codes from 'yum update' exit now if there is an issue.
 	if err != nil {
 		return nil, fmt.Errorf("error running yum with args %q: %v, stdout: %s", yumCheckUpdateArgs, err, out)
 	}
 
-	out, err = runWithPty(exec.Command(yum, yumListUpdatesArgs...))
-	DebugLogger.Printf("yum %q output:\n%s", yumListUpdatesArgs, strings.ReplaceAll(string(out), "\n", "\n "))
+	return parseAndUpdateYumPackages(opts...)
+}
+
+func parseAndUpdateYumPackages(opts ...YumUpdateOption) ([]PkgInfo, error) {
+	yumOpts := &yumUpdateOpts{
+		security: false,
+		minimal:  false,
+	}
+
+	for _, opt := range opts {
+		opt(yumOpts)
```

Comment:
This seems like a roundabout way to pass two booleans. Would it be clearer if this function, and `YumUpdates` take two params `security: bool, minimal: bool` instead?

Reply:
it is a general pattern that i have seen being used in golang code to pass around options. the same pattern in used for package managers as well.

Reply:
This is best practice for go options, you can add options without breaking the function interface.

Reply:
@ryanwe now i understand what you meant. i think i accidentally left the options parsing in original function that caused the confusion. it should be good now